data/youtube.js

```diff
@@ -24,159 +21,119 @@ axios.interceptors.response.use(response => {
+  }
+
+  return channelSectionPlaylistIDs;
+}
+
+/**
+ * Fetch playlists
+ * https://developers.google.com/youtube/v3/docs/playlists/list
+ * @param {String} channelSectionPlaylistIDs a playlist ID
+ */
+export async function fetchPlaylists(channelSectionPlaylistIDs) {
+  const playlists = channelSectionPlaylistIDs.map((id = '') => {
     return axios.get(`${url}/playlists`, {
       params: {
-        id: id,
-        part: PLAYLIST_PARTS.join(),
+        id,
         key: API_KEY,
+        part: 'id,snippet',
       },
     });
-  })
-  return await Promise.all(playlists).then((values) => {
-    return values
-    .map((playlist = {}) => {
-      const { data: { items }} = playlist
-      const { snippet: { publishedAt, id, title, thumbnails }} = items[0]
-      console.log(`fetched: Title: ${title} publishedAt: ${publishedAt}`)
+  });
+
+  const result = await Promise.all(playlists);
+
+  if (!result.length) {
+    throw new Error('No playlist data');
+  }
+
+  return result
+    .map(({ data: { items } } = {}) => {
+      const {
+        id,
+        snippet: { publishedAt, title, thumbnails },
+      } = items[0];
+
       return {
         publishDate: publishedAt,
         id,
         title,
-        thumbnails
-      }
+        thumbnails,
+      };
     })
-  })
+    .sort(sortByDate);
 }
 
-// const fetchPlaylists = async (nextPage, playlists = []) => {
-//   const { data = {} } =
-//     (await axios.get(`${url}/channelSections`, {
-//       params: {
-//         part: PLAYLIST_PARTS.join(),
-//         channelId: CHANNEL_ID,
-//         key: API_KEY,
-//         pageToken: nextPage,
-//         maxResults: 10,
-//       },
-//     })) || {};
-
-//   const { items = [], nextPageToken } = data;
-
-//   const morePlaylists = items
-//     .map((item = {}) => {
-//       const {
-//         snippet: { title } = {},
-//         contentDetails: { playlists }
-//       } = item;
-//       console.log("title = " + title)
-//       if (title != TARGET_SECTION_TITLE) {
-//         return false;
-//       }
-
-//       return {
-//         playlists
-//       };
-//     })
-//     .filter(Boolean);
-
-//   if (nextPageToken) {
-//     return fetchPlaylists(nextPageToken, [...playlists, ...morePlaylists]);
-//   }
-
-//   return [...playlists, ...morePlaylists];
-// };
-
-const fetchPlaylistItems = async (playlistId, nextPage, videos = [], recurse = true) => {
-  console.log("fetch id = " + playlistId)
+/**
+ * Fetch playlist videos
+ * https://developers.google.com/youtube/v3/docs/playlistItems/list
+ * @param {String} playlistId the playlist ID to get videos
+ */
+export async function fetchPlaylistItems(playlistId) {
   const { data = {} } =
     (await axios.get(`${url}/playlistItems`, {
       params: {
-        part: PLAYLIST_PARTS.join(),
-        playlistId,
         key: API_KEY,
-        pageToken: nextPage,
+        maxResults: 30, // ? 🤔 not sure the max # of videos in series playlists
```

Comment:
@dereckquock it's like 4 videos :)

Reply:
i think sometimes, there's like 7 videos 😅  i'm adding some buffer for times when maybe a series is a month long 😂